### PR TITLE
major dependency updates and revisions

### PR DESCRIPTION
**Pull request by zcapper**
#### Description

This PR revises our eslint config so its usable, working, and up to date

----
#### PR Type

- [ ] Bugfix or minor change _(1)_
- [ ] Feature _(2)_
- [X] Breaking change (existing functionality no longer works as expected) _(2)_
- [ ] Critical impact (security, payments or critical functionality) _(2+CTO)_

----
#### Changes

- Updates dependencies to the latest verions
- Adds peerDependencies as per eslint recommendations
- Adds documentation on usage and installation

----

#### Acceptance Criteria

_How can we test that this meets the requirements of the ticket?_

----

#### Checklist:

- [ ] Tests – Have you added or edited the test cases?
- [ ] Context – Have you tested this with a non-admin user?
- [ ] Rebase – Has this PR been rebased against `develop`?
- [ ] Feature Flag - If this is a new feature, does it have a feature flag?
- [ ] Release - Can this be released to production once tested?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
-    "eslint-plugin-react": "7.24.0",
+    "eslint-plugin-prettier": "4.2.1",
     "eslint-plugin-react-hooks": "^4",
-    "eslint-plugin-unicorn": "34.0.1",
-    "prettier": "2.8.3",
-    "typescript": "3.7"
+    "eslint-plugin-react": "7.32.1",
+    "eslint-plugin-unicorn": "45.0.2",
+    "eslint": "^8.32.0",
+    "typescript": "^4.9.4"
```
Comment (ish-bindra) [inline package.json]: Think it's a good opportunity to be flexible with version ranges now: https://github.com/npm/node-semver#caret-ranges-123-025-004
Especially with minor and patch updates.

**Review by ish-bindra — COMMENTED** [review]
I wonder what kind of follow on changes would this make? (Especially formatting and white space changes)
I couldn't test this myself because it isn't published afaict.

**Comment by zcapper** [comment]
@ish-bindra I've published it as @ordermentum/eslint-config-ordermentum@2.0.0-rc3
Import as `@ordermentum/*`, e.g. `@ordermentum/ordermentum` for the main ruleset.

**Comment by zcapper** [comment]
Have added `^`s to the versions

**Review by ish-bindra — APPROVED** [review]
🎊

**Pull request merged**